toolkit/tools/pkg/imagecustomizerlib/baseconfigs.go

```diff
@@ -0,0 +1,122 @@
+package imagecustomizerlib
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/microsoft/azure-linux-image-tools/toolkit/tools/imagecustomizerapi"
+	"github.com/microsoft/azure-linux-image-tools/toolkit/tools/internal/file"
+)
+
+type ConfigWithBasePath struct {
+	*imagecustomizerapi.Config
```

Comment:
nit: `Config *imagecustomizerapi.Config`

Omitting the variable name does a weird form on inheritance which I am not a fan of.